typhon/physics/em.py

```diff
@@ -358,3 +358,131 @@ def stefan_boltzmann_law(T):
         Energy per surface area [W m^-2]
     """
     return constants.stefan_boltzmann_constant * T**4
+
+
+def zeeman_splitting(gu, gl, mu, ml, H=1):
+    """ Zeeman splitting
+
+    .. math::
+        \Delta f = \\frac{H\mu_b}{h}(g_um_u - g_lm_l),
+    where $\mu_b$ is the Bohr magneton and $h$ is the Planck constant.
+
+    Parameters:
+        gu (scalar or ndarray) Upper g
+
+        gl (scalar or ndarray) Lower g
+
+        mu (scalar or ndarray) Upper projection of j
+
+        ml (scalar or ndarray) Lower projection of j
+
+        H (scalar or ndarray) Absolute strength of magnetic field in Teslas
+
+    Returns:
+        (scalar or ndarray) Splitting in Hertz
+    """
+    h = constants.planck
+    mu_B = constants.mu_B
+
+    frac = mu_B / h
+    return frac * (gu * mu - gl * ml) * H
+
+
+def zeeman_strength(ju, jl, mu, ml):
+    """ Zeeman line strength
+
+    .. math:: \Delta S_{M_u,M_l} = C \\left(\\begin{array}{ccc} J_l & 1 & J_u
+                       \\\\ M_l & M_u-M_l&-M_u \\end{array}\\right)^2,
+    where C is either 3/2 or 3/4 depending on in mu-ml is 0 or not.  In case
```

Comment:
There seems to be text missing here? It ends abruptly with 'In case'...

Reply:
Thanks!  Question:  This function requires sympy.  Is it OK to have it as a dependency?  A really useful package!

Reply:
Agreed that it is very useful. We haven't any other places where it's currently needed, therefore let's keep it as a soft dependency for now just like you implemented it.
Btw, there is no need to have all the code inside the try block. Put only the import statement into the try block and have the except block directly afterwards. All the code that uses sympy can then come after that. That way you save one level of indentation and make the code easier to read because the except block follows the try immediately.

```python
try:
        import sympy.physics.wigner as wig
except ModuleNotFoundError:
        raise RuntimeError("Must have sympy installed to use")

if np.isscalar(mu) and np.isscalar(ju):
# remaining code....
```

Reply:
The functions in the `typhon.physics.metrology` module all require sympy.  The module does not import sympy, but the functions all work on `sympy.Expr` objects, so although it does not introduce a dependency, the module is useless without sympy.